specification/columns/resourcetype.md

```diff
@@ -1,8 +1,8 @@
 # Resource Type
 
-Resource Type describes the kind of [*resource*](#glossary:resource) the charge applies to.  A Resource Type is commonly used for scenarios like identifying cost changes in groups of similar *resources* and may include values like Virtual Machine, Data Warehouse, and Load Balancer.
+Resource Type describes the kind of [*resource*](#glossary:resource) the charge applies to. A Resource Type is commonly used for scenarios like identifying cost changes in groups of similar *resources* and may include values like Virtual Machine, Data Warehouse, and Load Balancer.
 
-The ResourceType column MUST be present within billing data.  This column MUST be of type String and MUST NOT be null when a corresponding [ResourceId](#resourceid) is not null.  When a corresponding ResourceId value is null, the ResourceType column value MUST also be null.  Providers MUST use a consistent value-format and a set of values for ResourceType values within their billing datasets.
+The ResourceType column MUST be present in the billing data when the provider supports billing based on provisioned resources and supports the categorization of resources. This column MUST be of type String and MUST NOT be null when a corresponding [ResourceId](#resourceid) is not null. When a corresponding ResourceId value is null, the ResourceType column value MUST also be null. Providers MUST use a consistent value format and a set of values for ResourceType values within their billing datasets.
```

Comment:
nit: "Categorization" sounds like something different than the description of the type of resource.
```suggestion
The ResourceType column MUST be present in the billing data when the provider supports billing based on provisioned resources and supports more than one type of resource. This column MUST be of type String and MUST NOT be null when a corresponding [ResourceId](#resourceid) is not null. When a corresponding ResourceId value is null, the ResourceType column value MUST also be null. Providers MUST use a consistent value format and a set of values for ResourceType values within their billing datasets.
```

Reply:
Agree on the word categorization can be confused with our 'Category' columns. I struggled with these exclusion clauses. 

My question is how tightly do we tighten the screws here to make it required. We don't have a way to define what the scope of a 'type' is (what if a provider says storage volumes and virtual machines are both of resource type 'cloud resource'?).. do we communicate that it's almost up to the provider to decide unless they explicitly have a typing construct/mechanism in place? Maybe I'm overcomplicating this. Need a bit more convincing.. :)

Reply:
That's a good point... Someone could put "Resource" for everything 😒 Or someone could put "Windows Server 2024 Virtual Machine with Premium SSD" 😖 Given this is a "Type", I would be fine leaving it up to the provider and have practitioners push them for the right level of granularity, but either way "more than one type of resource" is still applicable. The question is more about whether we should be explicit about how granular "type" must be.

Reply:
but that means someone with just one category would possibly not provide or provide null? which isn't as helpful when aggregating across datasets. Don't want to hold up based on this. Are we ok with that semantic?

Can we go with something less specific:
".. when the provider supports billing based on provisioned resources and supports typing constructs for resources"
or this:
".. when the provider supports billing based on provisioned resources"

Reply:
Updated to:
The ResourceType column SHOULD be present in the billing data when the provider supports billing based on provisioned resources and supports assigning a type for resources.